src/components/NewCommentForm.tsx

```diff
@@ -45,24 +113,30 @@ export const NewCommentForm: React.FC = () => {
             name="email"
             id="comment-author-email"
             placeholder="email@test.com"
-            className="input is-danger"
+            className={cn('input', { 'is-danger': emailError })}
+            value={email}
+            onChange={e => setEmail(e.target.value)}
           />
 
-          <span className="icon is-small is-left">
-            <i className="fas fa-envelope" />
-          </span>
+          {!emailError ? (
+            <span className="icon is-small is-left">
+              <i className="fas fa-envelope" />
+            </span>
+          ) : (
+            <>
+              <span
+                className="icon is-small is-right has-text-danger"
+                data-cy="ErrorIcon"
+              >
+                <i className="fas fa-exclamation-triangle" />
+              </span>
 
-          <span
-            className="icon is-small is-right has-text-danger"
-            data-cy="ErrorIcon"
-          >
-            <i className="fas fa-exclamation-triangle" />
-          </span>
+              <p className="help is-danger" data-cy="ErrorMessage">
```

Comment:
make Error component which take `{error}` like a children props & reuse in all cases